### DISCUSSION TITLE
Two ssh keys

### DISCUSSION THREAD

**Discussion by Open-Sourcerer-Dev** [community/community#131257] [Code Security] [answered]
### Select Topic Area

Question

### Body

I have two github accounts, one aiming to contribute to open source and personal projects (this one I just made it) and another for my job and freelances. Now, Despite I have generated a ssh key in my machine then added it to both accounts, I'm having issues when interacting with my private repos, even the ones which was working.

**Comment by GabrielLins64** [comment, marked as answer]
Supose you have the keys `id_rsa_personal` and `id_rsa_job`, then you'll have to add them both to the ssh agent

```shell
ssh-add ~/.ssh/id_rsa_personal
ssh-add ~/.ssh/id_rsa_job
```

Create a ssh config for managing both keys/accounts:

```shell
touch ~/.ssh/config
```

Add this content to the config file:

```cfg
Host gh_job
    HostName github.com
    IdentityFile ~/.ssh/id_rsa_job

Host gh_personal
    HostName github.com
    IdentityFile ~/.ssh/id_rsa_personal
```

Now, supose you'll want to clone a repository from your personal account, you'll have to change a part of the url like this:

Change
`git clone git@github.com:your-account/your-repository.git`
To
`git clone git@gh_personal:your-account/your-repository.git`

Similarly, if you already have a repository, of your job account for example, just change where its upstream/origin points:

```shell
git remote set-url origin git@gh_job:your-another-account/your-repository.git
```
